### PR TITLE
[cleanup] move non-reconcile functions into controller helpers

### PR DESCRIPTION
**Pull request by AshleyDumaine**
**What this PR does / why we need it**: General cleanup for controllers controller which should have helper functions in the appropriate helpers file for consistency

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**TODOs**:

- [x] squashed commits
- [ ] includes documentation
- [ ] adds unit tests
- [ ] adds or updates e2e tests

### CODE REVIEW

**Review by komer3 — DISMISSED** [review]
LGTM!

File: controller/linodecluster_controller_helpers.go [hunk 1]
```diff
@@ -40,3 +51,110 @@ func (r *LinodeClusterReconciler) removeMachineFromLB(ctx context.Context, logge
 	}
 	return nil
 }
+
```
Comment (amold1) [inline controller/linodecluster_controller_helpers.go:54]: can we make this a "common" function instead on a cluster reconciler method?
The only thing it uses from the reconciler is the TracedClient. If we just pass the client, other reconcilers could use the same function like [this one](https://github.com/linode/cluster-api-provider-linode/blob/main/controller/linodemachine_controller_helpers.go#L191)

File: controller/linodecluster_controller_helpers.go [hunk 2]
```diff
@@ -40,3 +51,110 @@ func (r *LinodeClusterReconciler) removeMachineFromLB(ctx context.Context, logge
...
+				Namespace: linodeCluster.Namespace,
+				Name:      linodeCluster.Name,
+			},
+		})
+
+		return result
+	}
+}
+
+func (r *LinodeClusterReconciler) handleDNS(clusterScope *scope.ClusterScope) {
```
Comment (amold1) [inline controller/linodecluster_controller_helpers.go:98]: looks like all these functions could do without being a reconciler method

File: controller/linodemachine_controller.go [hunk 3]
```diff
@@ -125,25 +118,38 @@ func (r *LinodeMachineReconciler) Reconcile(ctx context.Context, req ctrl.Reques
...
+	cluster, err := kutil.GetClusterFromMetadata(ctx, r.TracedClient(), machine.ObjectMeta)
 	if err != nil || cluster == nil {
 		return ctrl.Result{}, err
 	}
 
+	linodeClusterKey := client.ObjectKey{
+		Namespace: linodeMachine.Namespace,
+		Name:      cluster.Spec.InfrastructureRef.Name,
+	}
+	linodeCluster := &infrav1alpha2.LinodeCluster{}
```
Comment (amold1) [inline controller/linodemachine_controller.go:136]: I like this approach. But now that we are populating the field here, we should remove the `Get()` calls in the reconcile method

**Review by amold1 — COMMENTED** [review]
we should probably create a generic/common helpers file and put the non-reconciler specific functions in there

**Comment by AshleyDumaine** [comment]
> we should probably create a generic/common helpers file and put the non-reconciler specific functions in there

since we pass machineScope around so much, it looks like nothing actually needs the reconciler (`r.Get` can be replaced with `machineScope.Client.Get`). I was able to remove it on everything.

File: controller/linodecluster_controller_helpers.go (hunk 1 above)
Comment (AshleyDumaine) [reply controller/linodecluster_controller_helpers.go:54]: updated to pass the traced client

**Review by amold1 — DISMISSED** [review]

**Comment by amold1** [comment]
LGTM

File: controller/linodecluster_controller_helpers.go (hunk 2 above)
Comment (AshleyDumaine) [reply controller/linodecluster_controller_helpers.go]: updated

**Review by amold1 — DISMISSED** [review]

**Review by eljohnson92 — APPROVED** [review]

**Review by eljohnson92 — APPROVED** [review]

**Pull request merged**